velox/exec/tests/utils/DistributedPlanBuilder.cpp

```diff
@@ -108,6 +108,29 @@ core::PlanNodePtr DistributedPlanBuilder::shuffleResult(
   return std::move(planNode_);
 }
 
+core::PlanNodePtr DistributedPlanBuilder::broadcastResult() {
+  partitionedOutputBroadcast();
+  auto* output =
+      dynamic_cast<const core::PartitionedOutputNode*>(planNode_.get());
+  VELOX_CHECK_NOT_NULL(output);
+  const auto producerPrefix = current_->taskPrefix;
+  auto result = planNode_;
+  newFragment();
+  root_->stack_.pop_back();
+  auto* consumer = root_->stack_.back();
+  fragments_.back().numBroadcastDestinations = consumer->current_->width;
+  VELOX_CHECK_GE(consumer->current_->width, 1);
```

Comment:
We should check width before its use, and we shall also check numBroadcastDestinations is not set in the previous fragment yet.